core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -64,6 +68,25 @@ pub struct Cast {
     pub timezone: String,
 }
 
+macro_rules! spark_cast_utf8_to_integral {
```

Comment:
maybe `utf8_to_integer`? 

spark not involved in native exec, not sure why spark is needed.
Integral type also includes booleans and this scope limited by integers afaik